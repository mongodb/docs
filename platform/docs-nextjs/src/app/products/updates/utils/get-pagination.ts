/**
 *
 * @param array - flatten array
 * @param chunkSize - max size of each array
 * @returns [[{}, {}], [{}]]
 * @description Takes an flatten array and creates one level nested array of objects based on the chunkSize. Used for pagination data.
 */
export const getPagination = <T>(array: T[], chunkSize: number): T[][] => {
  const chunks: T[][] = [];
  if (!array.length) {
    return [[]]; // default to one empty nested array
  }
  for (let i = 0; i < array.length; i += chunkSize) {
    chunks.push(array.slice(i, i + chunkSize));
  }
  return chunks;
};
