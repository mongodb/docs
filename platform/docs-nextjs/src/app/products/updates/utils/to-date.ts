export const normalizeDate = (date: string) => {
  // Check if the date string has the correct format 'YYYY-MM-DD'
  const dateToNormalize = /^(\d{4})-(\d{2})-(\d{2})$/.exec(date)?.slice(1); // excludes the full date in the first index i.e. "2025-05-27"
  if (dateToNormalize) {
    // This is needed because, JavaScript’s Date constructor uses zero-based months (January = 0 … December = 11).
    return new Date(Number(dateToNormalize[0]), Number(dateToNormalize[1]) - 1, Number(dateToNormalize[2]));
  }
  return new Date(date);
};
