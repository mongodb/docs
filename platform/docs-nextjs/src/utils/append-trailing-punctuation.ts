import { ASTNode, ReferenceNode, RefRoleNode } from "@/types/ast";
import { isReferenceNode, isRefRoleNode, isTextNode } from "@/types/ast-utils";

// Append dangling punctuation to preceding Link's text
export const appendTrailingPunctuation = (nodes: ASTNode[]) => {
  const truncatedNodes = [];

  for (let i = 0; i < nodes.length; i++) {
    const currNode = nodes[i];
    const nextNode = nodes[i + 1];
    const hasDanglingSibling =
      isTextNode(nextNode) &&
      nextNode.value.length === 1 &&
      !nextNode.value.match(/\s/);
    // TODO: if issues arise, instead of matching for non-whitespace, we might consider checking FOR certain singular punctuation
    // ie: regex match for  .?!,:;)*"']}`

    if ((isReferenceNode(currNode) || isRefRoleNode(currNode)) && hasDanglingSibling) {
      const copyOfNodeWithDeepCopiedChildren = deepCopyNodesChildren(currNode);
      copyOfNodeWithDeepCopiedChildren.children.push(nextNode);
      truncatedNodes.push(copyOfNodeWithDeepCopiedChildren);
      i++;
      continue;
    }
    truncatedNodes.push(currNode);
  }
  return truncatedNodes;
};

// Make a copy of node with a deep copy of the new child node
function deepCopyNodesChildren(refNode: ReferenceNode | RefRoleNode) {
  if (!refNode.children || !refNode.children.length) return { ...refNode };
  const copyOfChildren = [...refNode.children];

  const copyOfNode = {
    ...refNode,
    children: copyOfChildren,
  };
  return copyOfNode;
}
