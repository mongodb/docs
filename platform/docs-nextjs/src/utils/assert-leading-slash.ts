const assertLeadingSlash = (url: string) => {
  if (url && url.match(/^\//)) {
    return url;
  }
  return `/${url}`;
};

export { assertLeadingSlash };
