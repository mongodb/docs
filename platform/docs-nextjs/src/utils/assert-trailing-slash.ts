export const assertTrailingSlash = (url: string) => {
  if (url && url.match(/\/$/)) {
    return url;
  }
  return `${url}/`;
};
