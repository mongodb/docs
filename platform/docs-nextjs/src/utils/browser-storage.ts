import { isBrowser } from './is-browser';

const isValidStorage = isBrowser;

export const setLocalValue = (key: string, value: unknown) => {
  try {
    if (isValidStorage) {
      const prevState = JSON.parse(window.localStorage.getItem('mongodb-docs') || '{}');
      localStorage.setItem('mongodb-docs', JSON.stringify({ ...prevState, [key]: value }));
    }
  } catch {
    console.error('Error setting localStorage value');
  }
};

export const getLocalValue = (key: string) => {
  try {
    if (isValidStorage && JSON.parse(window.localStorage.getItem('mongodb-docs') || '{}')) {
      const docsObj = JSON.parse(window.localStorage.getItem('mongodb-docs') || '{}');
      if (docsObj) {
        return docsObj[key];
      }
    }
    return undefined;
  } catch {
    console.error('Error getting localStorage value');
    return undefined;
  }
};

export const setSessionValue = (key: string, value: string) => {
  try {
    if (isValidStorage) {
      const prevState = JSON.parse(window.sessionStorage.getItem('mongodb-docs') || '{}');
      sessionStorage.setItem('mongodb-docs', JSON.stringify({ ...prevState, [key]: value }));
    }
  } catch {
    console.error('Error setting session value');
  }
};

export const getSessionValue = (key: string) => {
  try {
    if (isValidStorage && JSON.parse(window.sessionStorage.getItem('mongodb-docs') || '{}')) {
      const docsObj = JSON.parse(window.sessionStorage.getItem('mongodb-docs') || '{}');
      if (docsObj) {
        return docsObj[key];
      }
    }
    return undefined;
  } catch {
    console.error('Error getting session value');
    return undefined;
  }
};
