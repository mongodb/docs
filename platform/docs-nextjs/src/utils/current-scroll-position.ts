import { isBrowser } from './is-browser';

// Function that returns current scroll position for analytics
export const currentScrollPosition = (): number => {
  if (isBrowser) {
    const percentage = Math.round((window.scrollY / (document.body.scrollHeight - window.innerHeight)) * 100);
    return Math.round(percentage / 10) * 10;
  }
  return 0;
};
