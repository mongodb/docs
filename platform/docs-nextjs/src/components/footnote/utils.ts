import type { ASTNode } from '@/types/ast';
import type { FootnoteNode, FootnoteReferenceNode } from '@/types/ast';
import { findAllKeyValuePairs } from '@/utils/find-all-key-value-pairs';
import type { FootnoteType } from './footnote-context';

export const getFootnotes = (nodes: ASTNode[]) => {
  const footnotes = findAllKeyValuePairs(nodes, 'type', 'footnote') as FootnoteNode[];
  const footnoteReferences = findAllKeyValuePairs(nodes, 'type', 'footnote_reference') as FootnoteReferenceNode[];
  const numAnonRefs = footnoteReferences.filter(
    (node) => !Object.prototype.hasOwnProperty.call(node, 'refname'),
  ).length;
  // We label our footnotes by their index, regardless of their names to
  // circumvent cases such as [[1], [#], [2], ...]
  return footnotes.reduce<Record<string, FootnoteType>>((map, footnote, index) => {
    if (footnote.name) {
      // Find references associated with a named footnote

      map[footnote.name] = {
        label: index + 1,
        references: getNamedFootnoteReferences(footnoteReferences, footnote.name),
      };
    } else {
      // Find references associated with an anonymous footnote
      // Replace potentially broken anonymous footnote ids
      footnote.id = `${index + 1}`;

      map[footnote.id] = {
        label: index + 1,
        references: getAnonymousFootnoteReferences(index, numAnonRefs),
      };
    }
    return map;
  }, {});
};

// Find all footnote_reference node IDs associated with a given footnote by
// that footnote's refname
const getNamedFootnoteReferences = (footnoteReferences: FootnoteReferenceNode[], refname: string) => {
  return footnoteReferences.filter((node) => node.refname === refname).map((node) => node.id);
};

// They are used infrequently, but here we match an anonymous footnote to its reference.
// The nth footnote on a page is associated with the nth reference on the page. Since
// anon footnotes and footnote references are anonymous, we assume a 1:1 pairing, and
// have no need to query nodes. If there are more anonymous footnotes than references,
// we may return an empty array
const getAnonymousFootnoteReferences = (index: number, numAnonRefs: number) => {
  return index > numAnonRefs ? [] : [`id${index + 1}`];
};
