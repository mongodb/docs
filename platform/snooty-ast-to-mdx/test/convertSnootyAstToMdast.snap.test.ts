import type { SnootyNode } from '../src/core/convertSnootyAstToMdast/types';
import { convertSnootyAst } from './utils';

/** Minimal Snooty AST objects to snapshot test the MDX conversion against */
const cases: Array<[string, SnootyNode]> = [
  ['simple text node', { type: 'text', value: 'Just text' }],
  [
    'paragraph',
    {
      type: 'paragraph',
      children: [{ type: 'text', value: 'Hello world' }],
    },
  ],
  [
    'emphasis',
    {
      type: 'paragraph',
      children: [
        {
          type: 'emphasis',
          children: [{ type: 'text', value: 'em' }],
        },
      ],
    },
  ],
  [
    'strong',
    {
      type: 'paragraph',
      children: [
        {
          type: 'strong',
          children: [{ type: 'text', value: 'bold' }],
        },
      ],
    },
  ],
  [
    'inline literal â†’ inline code',
    {
      type: 'paragraph',
      children: [{ type: 'literal', value: 'code' }],
    },
  ],
  [
    'code block',
    {
      type: 'code',
      lang: 'js',
      value: 'console.log("hi")',
    },
  ],
  [
    'bullet_list',
    {
      type: 'bullet_list',
      children: [
        {
          type: 'list_item',
          children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Item' }] }],
        },
      ],
    },
  ],
  [
    'ordered_list',
    {
      type: 'ordered_list',
      start: 1,
      children: [
        {
          type: 'list_item',
          children: [{ type: 'paragraph', children: [{ type: 'text', value: 'First' }] }],
        },
      ],
    },
  ],
  [
    'link reference',
    {
      type: 'reference',
      refuri: 'https://example.com',
      children: [{ type: 'text', value: 'link' }],
    },
  ],
  [
    'heading',
    {
      type: 'heading',
      depth: 2,
      children: [{ type: 'text', value: 'Heading' }],
    },
  ],
  [
    'blockquote',
    {
      type: 'block_quote',
      children: [
        {
          type: 'paragraph',
          children: [{ type: 'text', value: 'quote' }],
        },
      ],
    },
  ],
  [
    'field list',
    {
      type: 'field_list',
      children: [
        {
          type: 'field',
          name: 'foo',
          label: 'Foo',
          children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Field content' }] }],
        },
      ],
    },
  ],
  [
    'superscript and subscript',
    {
      type: 'paragraph',
      children: [
        { type: 'superscript', children: [{ type: 'text', value: 'sup' }] },
        { type: 'text', value: ' and ' },
        { type: 'subscript', children: [{ type: 'text', value: 'sub' }] },
      ],
    },
  ],
  [
    'definition list',
    {
      type: 'definitionList',
      children: [
        {
          type: 'definitionListItem',
          term: [{ type: 'text', value: 'Term' }],
          children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Definition' }] }],
        },
      ],
    },
  ],
  [
    'table',
    {
      type: 'table',
      children: [
        {
          type: 'table_row',
          children: [{ type: 'table_cell', children: [{ type: 'text', value: 'Cell' }] }],
        },
      ],
    },
  ],
  [
    'directive image',
    {
      type: 'directive',
      name: 'image',
      argument: '/images/test.png',
    },
  ],
  [
    'directive literalinclude',
    {
      type: 'directive',
      name: 'literalinclude',
      options: { language: 'js' },
      argument: 'src/code.js',
    },
  ],
  [
    'directive include',
    {
      type: 'directive',
      name: 'include',
      argument: 'included-file',
      children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Included' }] }],
    },
  ],
  [
    'only directive',
    {
      type: 'directive',
      name: 'only',
      argument: 'target == "mytarget"',
      children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Only content' }] }],
    },
  ],
  [
    'role abbr',
    {
      type: 'role',
      name: 'abbr',
      value: 'abbr',
      children: [],
    },
  ],
  [
    'admonition note',
    {
      type: 'admonition',
      name: 'note',
      children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Note text' }] }],
    },
  ],
  [
    'tabs container',
    {
      type: 'tabs',
      children: [
        {
          type: 'paragraph',
          children: [{ type: 'text', value: 'Tab content' }],
        },
      ],
    },
  ],
  [
    'target node',
    {
      type: 'target',
      ids: ['my-target'],
    },
  ],
  [
    'thematic break',
    {
      type: 'transition',
    },
  ],
  [
    'line block',
    {
      type: 'line_block',
      children: [
        { type: 'line', value: 'First line' },
        { type: 'line', value: 'Second line' },
      ],
    },
  ],
  [
    'footnote definition and reference',
    {
      type: 'root',
      children: [
        {
          type: 'paragraph',
          children: [
            { type: 'text', value: 'See note' },
            { type: 'footnote_reference', id: 1 },
          ],
        },
        {
          type: 'footnote',
          id: 1,
          children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Footnote text' }] }],
        },
      ],
    },
  ],
  [
    'substitution reference',
    {
      type: 'root',
      children: [
        {
          type: 'paragraph',
          children: [
            { type: 'text', value: 'Hello ' },
            { type: 'substitution_reference', refname: 'product', children: [{ type: 'text', value: 'MongoDB' }] },
          ],
        },
      ],
    },
  ],
  [
    'card-group',
    {
      type: 'card-group',
      options: { cols: 3 },
      children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Card' }] }],
    },
  ],
  [
    'cta-banner',
    {
      type: 'cta-banner',
      options: { color: 'green' },
      children: [{ type: 'paragraph', children: [{ type: 'text', value: 'CTA' }] }],
    },
  ],
  [
    'method-selector',
    {
      type: 'method-selector',
      children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Methods' }] }],
    },
  ],
  [
    'section with title',
    {
      type: 'section',
      children: [
        { type: 'title', children: [{ type: 'text', value: 'Section Title' }] },
        { type: 'paragraph', children: [{ type: 'text', value: 'Section body' }] },
      ],
    },
  ],
  [
    'generic ordered list',
    {
      type: 'list',
      enumtype: 'ordered',
      startat: 3,
      children: [{ type: 'listItem', children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Third' }] }] }],
    },
  ],
  [
    'doc link',
    {
      type: 'doc',
      url: '/docs/page',
      children: [{ type: 'text', value: 'Doc Page' }],
    },
  ],
  [
    'ref_role link',
    {
      type: 'ref_role',
      url: '/ref/link',
      children: [{ type: 'text', value: 'Ref Link' }],
    },
  ],
  [
    'directive cond',
    {
      type: 'directive',
      name: 'cond',
      argument: 'env=="production"',
      children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Conditional' }] }],
    },
  ],
  [
    'inline_target',
    {
      type: 'inline_target',
      ids: ['inline-id'],
    },
  ],
  [
    'enumerated_list',
    {
      type: 'enumerated_list',
      start: 5,
      children: [{ type: 'list_item', children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Five' }] }] }],
    },
  ],
  [
    'standalone list_item',
    {
      type: 'list_item',
      children: [{ type: 'paragraph', children: [{ type: 'text', value: 'Alone item' }] }],
    },
  ],
  [
    'table head',
    {
      type: 'table_head',
      children: [
        {
          type: 'table_row',
          children: [
            {
              type: 'table_cell',
              children: [{ type: 'text', value: 'Header' }],
            },
          ],
        },
      ],
    },
  ],
  [
    'title reference',
    {
      type: 'title_reference',
      children: [{ type: 'text', value: 'Titled' }],
    },
  ],
  [
    'substitution node',
    {
      type: 'substitution',
      refname: 'product',
      children: [{ type: 'text', value: 'MongoDB' }],
    },
  ],
  ['hidden directive (toctree)', { type: 'directive', name: 'toctree' }],
  ['unknown node type', { type: 'mystery' }],
  ['title node', { type: 'title', children: [{ type: 'text', value: 'A Title' }] }],
  ['directive figure', { type: 'directive', name: 'figure', argument: 'images/figure.png' }],
  ['directive sharedinclude', { type: 'directive', name: 'sharedinclude', argument: 'shared-file.rst' }],
  ['directive_argument node', { type: 'directive_argument', children: [{ type: 'text', value: 'Arg text' }] }],
  ['comment node skipped', { type: 'comment', value: 'This is a comment' }],
  ['comment_block node skipped', { type: 'comment_block', value: 'Block comment' }],
  [
    'named_reference node',
    { type: 'named_reference', refname: 'named', children: [{ type: 'text', value: 'NamedRef' }] },
  ],
  [
    'substitution_definition node',
    { type: 'substitution_definition', refname: 'prod', children: [{ type: 'text', value: 'MongoDB' }] },
  ],
  ['target_identifier', { type: 'target_identifier', ids: ['ident'] }],
  [
    'directive list-table',
    {
      type: 'directive',
      name: 'list-table',
      options: { 'header-rows': 1 },
      children: [
        {
          type: 'bullet_list',
          children: [
            {
              type: 'list_item',
              children: [
                {
                  type: 'bullet_list',
                  children: [
                    {
                      type: 'list_item',
                      children: [
                        {
                          type: 'paragraph',
                          children: [{ type: 'text', value: 'Version' }],
                        },
                      ],
                    },
                    {
                      type: 'list_item',
                      children: [
                        {
                          type: 'paragraph',
                          children: [{ type: 'text', value: 'client 1.0' }],
                        },
                      ],
                    },
                    {
                      type: 'list_item',
                      children: [
                        {
                          type: 'paragraph',
                          children: [{ type: 'text', value: 'client 2.0' }],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
            {
              type: 'list_item',
              children: [
                {
                  type: 'bullet_list',
                  children: [
                    {
                      type: 'list_item',
                      children: [
                        {
                          type: 'paragraph',
                          children: [{ type: 'text', value: 'DB 7.0' }],
                        },
                      ],
                    },
                    {
                      type: 'list_item',
                      children: [
                        {
                          type: 'paragraph',
                          children: [{ type: 'text', value: 'âœ“' }],
                        },
                      ],
                    },
                    {
                      type: 'list_item',
                      children: [
                        {
                          type: 'paragraph',
                          children: [{ type: 'text', value: 'âœ“' }],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
            {
              type: 'list_item',
              children: [
                {
                  type: 'bullet_list',
                  children: [
                    {
                      type: 'list_item',
                      children: [
                        {
                          type: 'paragraph',
                          children: [{ type: 'text', value: 'DB 8.0' }],
                        },
                      ],
                    },
                    {
                      type: 'list_item',
                      children: [
                        {
                          type: 'paragraph',
                          children: [{ type: 'text', value: 'âœ“' }],
                        },
                      ],
                    },
                    {
                      type: 'list_item',
                      children: [
                        {
                          type: 'paragraph',
                          children: [{ type: 'text', value: '-' }],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
  ],
];

describe('convertSnootyAstToMdast - node snapshots', () => {
  test.each(cases)('| %s', (_, ast) => {
    const { mdx } = convertSnootyAst({ ast });
    expect(mdx).toMatchSnapshot();
  });
});
