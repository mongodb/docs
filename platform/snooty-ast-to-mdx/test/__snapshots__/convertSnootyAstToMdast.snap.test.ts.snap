// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`convertSnootyAstToMdast - node snapshots | admonition note 1`] = `
"<Note>
  Note text
</Note>"
`;

exports[`convertSnootyAstToMdast - node snapshots | blockquote 1`] = `"> quote"`;

exports[`convertSnootyAstToMdast - node snapshots | bullet_list 1`] = `"* Item"`;

exports[`convertSnootyAstToMdast - node snapshots | card-group 1`] = `
"<CardGroup cols={3}>
  Card
</CardGroup>"
`;

exports[`convertSnootyAstToMdast - node snapshots | code block 1`] = `
"\`\`\`js
console.log("hi")
\`\`\`"
`;

exports[`convertSnootyAstToMdast - node snapshots | comment node skipped 1`] = `""`;

exports[`convertSnootyAstToMdast - node snapshots | comment_block node skipped 1`] = `""`;

exports[`convertSnootyAstToMdast - node snapshots | cta-banner 1`] = `
"<CTABanner color="green">
  CTA
</CTABanner>"
`;

exports[`convertSnootyAstToMdast - node snapshots | definition list 1`] = `
"<DefinitionList>
  <DefinitionListItem>
    Term

    Definition
  </DefinitionListItem>
</DefinitionList>"
`;

exports[`convertSnootyAstToMdast - node snapshots | directive cond 1`] = `
"<Cond expr="env==&#x22;production&#x22;">
  Conditional
</Cond>"
`;

exports[`convertSnootyAstToMdast - node snapshots | directive figure 1`] = `
"import FigureImg from '../images/figure.png';

<Image src={FigureImg} />"
`;

exports[`convertSnootyAstToMdast - node snapshots | directive image 1`] = `
"import TestImg from '../images/test.png';

<Image src={TestImg} />"
`;

exports[`convertSnootyAstToMdast - node snapshots | directive include 1`] = `
"import IncludedFile from './included-file.mdx';

<IncludedFile />"
`;

exports[`convertSnootyAstToMdast - node snapshots | directive list-table 1`] = `
"| Version | client 1.0 | client 2.0 |
| ------- | ---------- | ---------- |
| DB 7.0  | ✓          | ✓          |
| DB 8.0  | ✓          | -          |"
`;

exports[`convertSnootyAstToMdast - node snapshots | directive literalinclude 1`] = `
"\`\`\`js
// Source: src/code.js
// TODO: Content from external file not available during conversion
\`\`\`"
`;

exports[`convertSnootyAstToMdast - node snapshots | directive sharedinclude 1`] = `
"import SharedFile from './shared-file.mdx';

<SharedFile />"
`;

exports[`convertSnootyAstToMdast - node snapshots | directive_argument node 1`] = `"Arg text"`;

exports[`convertSnootyAstToMdast - node snapshots | doc link 1`] = `"import { refs } from './references';<Ref url="/docs/page">{refs["/docs/page"].title}</Ref>"`;

exports[`convertSnootyAstToMdast - node snapshots | emphasis 1`] = `"*em*"`;

exports[`convertSnootyAstToMdast - node snapshots | enumerated_list 1`] = `"5. Five"`;

exports[`convertSnootyAstToMdast - node snapshots | field list 1`] = `
"<FieldList>
  <Field name="foo" label="Foo">
    Field content
  </Field>
</FieldList>"
`;

exports[`convertSnootyAstToMdast - node snapshots | footnote definition and reference 1`] = `
"See note[^1]

[^1]: Footnote text"
`;

exports[`convertSnootyAstToMdast - node snapshots | generic ordered list 1`] = `"3. Third"`;

exports[`convertSnootyAstToMdast - node snapshots | heading 1`] = `"## Heading"`;

exports[`convertSnootyAstToMdast - node snapshots | hidden directive (toctree) 1`] = `""`;

exports[`convertSnootyAstToMdast - node snapshots | inline literal → inline code 1`] = `"\`code\`"`;

exports[`convertSnootyAstToMdast - node snapshots | inline_target 1`] = `"<span id="inline-id" />"`;

exports[`convertSnootyAstToMdast - node snapshots | line block 1`] = `
"First line\\
Second line"
`;

exports[`convertSnootyAstToMdast - node snapshots | link reference 1`] = `"[link](https://example.com)"`;

exports[`convertSnootyAstToMdast - node snapshots | method-selector 1`] = `
"<MethodSelector>
  Methods
</MethodSelector>"
`;

exports[`convertSnootyAstToMdast - node snapshots | named_reference node 1`] = `""`;

exports[`convertSnootyAstToMdast - node snapshots | only directive 1`] = `
"<Only expr="target == &#x22;mytarget&#x22;">
  Only content
</Only>"
`;

exports[`convertSnootyAstToMdast - node snapshots | ordered_list 1`] = `"1. First"`;

exports[`convertSnootyAstToMdast - node snapshots | paragraph 1`] = `"Hello world"`;

exports[`convertSnootyAstToMdast - node snapshots | ref_role link 1`] = `"import { refs } from './references';<Ref url="/ref/link">{refs["/ref/link"].title}</Ref>"`;

exports[`convertSnootyAstToMdast - node snapshots | role abbr 1`] = `"<Abbr>abbr</Abbr>"`;

exports[`convertSnootyAstToMdast - node snapshots | section with title 1`] = `
"# Section Title

Section body"
`;

exports[`convertSnootyAstToMdast - node snapshots | simple text node 1`] = `"Just text"`;

exports[`convertSnootyAstToMdast - node snapshots | standalone list_item 1`] = `"* Alone item"`;

exports[`convertSnootyAstToMdast - node snapshots | strong 1`] = `"**bold**"`;

exports[`convertSnootyAstToMdast - node snapshots | substitution node 1`] = `"import { substitutions } from './references';{substitutions["product"]}"`;

exports[`convertSnootyAstToMdast - node snapshots | substitution reference 1`] = `
"import { substitutions } from './references';

Hello {substitutions["product"]}"
`;

exports[`convertSnootyAstToMdast - node snapshots | substitution_definition node 1`] = `""`;

exports[`convertSnootyAstToMdast - node snapshots | superscript and subscript 1`] = `"<sup>sup</sup> and <sub>sub</sub>"`;

exports[`convertSnootyAstToMdast - node snapshots | table 1`] = `
"<Table>
  <TableRow>
    <TableCell>
      Cell
    </TableCell>
  </TableRow>
</Table>"
`;

exports[`convertSnootyAstToMdast - node snapshots | table head 1`] = `
"<TableHead>
  <TableRow>
    <TableCell>
      Header
    </TableCell>
  </TableRow>
</TableHead>"
`;

exports[`convertSnootyAstToMdast - node snapshots | tabs container 1`] = `
"<Tabs>
  Tab content
</Tabs>"
`;

exports[`convertSnootyAstToMdast - node snapshots | target node 1`] = `"<span id="my-target" />"`;

exports[`convertSnootyAstToMdast - node snapshots | target_identifier 1`] = `"<span id="ident" />"`;

exports[`convertSnootyAstToMdast - node snapshots | thematic break 1`] = `"***"`;

exports[`convertSnootyAstToMdast - node snapshots | title node 1`] = `"# A Title"`;

exports[`convertSnootyAstToMdast - node snapshots | title reference 1`] = `"*Titled*"`;

exports[`convertSnootyAstToMdast - node snapshots | unknown node type 1`] = `"<!-- unsupported: mystery -->"`;
